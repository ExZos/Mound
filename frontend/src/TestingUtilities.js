import GeneralComponent from './components/GeneralComponent';

class TestingUtilities extends GeneralComponent {
  addUserSessionItem(userID, spaceID) {
    const user = {
      id: userID,
      name: 'name' + userID,
      space: spaceID,
      space_name: 'space' + spaceID,
      space_status: true
    };

    this.addToSessionArrayItem('users', user);
  }
}

export default TestingUtilities;
