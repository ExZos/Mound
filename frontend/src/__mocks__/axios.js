import { endpoints as api } from '../endpoints';

const axios = {
  create: () => axios,
  get: (url) => {
    switch(url) {
      case api.getSpaceByName + "Headon's Floor/":
        return Promise.resolve({
          data: {
            id: 1,
            name: "Headon's Floor",
            status: true
          }
        });
        break;

      default:
        return Promise.reject();
    }
  }
};

export default axios;
