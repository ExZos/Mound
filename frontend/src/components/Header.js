import React from 'react';
import { Link } from 'react-router-dom';

import GeneralComponent from './GeneralComponent';
import '../styles/header.css';

class Header extends GeneralComponent {
  // TODO: new message indicators
  renderSpaces = () => {
    const users = this.getSessionItem('users');

    if(users) {
      return Object.values(users).map(user => (
        <span key={user.space} className="spaceNavItem ">
          <Link className={(user.space === this.props.spaceID) ? "active" : ""} tabIndex="-1" to={{
            pathname: "/r/",
            state: {
              pathname: "/s/",
              state: {
                space: {
                  id: user.space,
                  name: user.space_name,
                  status: user.space_status
                }
              }
            }
          }}>
            {user.space_name}
          </Link>
        </span>
      ));
    }
  }

  render() {
    return(
      <div id="header">
        <div id="menuNav">
          <span className="menuItem">
            <Link to="/" tabIndex="-1">
              Home
            </Link>
          </span>

          <span className="menuItem">
            <Link to="/" className="clearSesh" tabIndex="-1" onClick={this.clearSession}>CLEAR</Link>
          </span>
        </div>

        <div id="spaceNav">
          {this.renderSpaces()}
        </div>
      </div>
    );
  }
}

export default Header;
