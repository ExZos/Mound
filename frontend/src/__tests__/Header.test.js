import React from 'react';
import { shallow, mount, render } from 'enzyme';
import axios from 'axios';
import { createBrowserHistory } from 'history';

import Header from '../components/Header';
import TestingUtilities from '../TestingUtilities';

const history = createBrowserHistory();
const tutils = new TestingUtilities();

describe('Header', () => {
  beforeEach(() => {
    sessionStorage.clear();
  });

  it('should render correctly without props nor session', () => {
    const component = shallow(<Header />);
    const tabs = component.find('span.spaceNavItem');

    expect(component).toMatchSnapshot();
    expect(tabs.length).toEqual(0);
  });

  it('should render tabs with session storage', () => {
    tutils.addUserSessionItem(1, 1);
    tutils.addUserSessionItem(2, 2);
    tutils.addUserSessionItem(3, 3);

    const component = shallow(<Header />);
    const tabs = component.find('span.spaceNavItem');

    expect(component).toMatchSnapshot();
    expect(tabs.length).toEqual(3);
  });

  it('should highlight a tab with props and session', () => {
    tutils.addUserSessionItem(1, 1);
    tutils.addUserSessionItem(2, 2);
    tutils.addUserSessionItem(3, 3);

    const component = shallow(<Header spaceID={2} />);
    const tab = component.find('Link.active');

    expect(tab.length).toEqual(1);
    expect(tab.children().text()).toEqual("space2");
  });

  it('should clear session on clear click', () => {
    tutils.addUserSessionItem(1, 1);
    tutils.addUserSessionItem(2, 2);
    tutils.addUserSessionItem(3, 3);

    const component = shallow(<Header />);
    const clear = component.find('Link.clearSesh');

    clear.props().onClick();

    expect(sessionStorage.getItem('users')).not.toBeDefined;
  });

  it('should ', () => {

  });
});
