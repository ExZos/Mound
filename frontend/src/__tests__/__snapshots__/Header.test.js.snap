// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly without props nor session 1`] = `
<div
  id="header"
>
  <div
    id="menuNav"
  >
    <span
      className="menuItem"
    >
      <Link
        tabIndex="-1"
        to="/"
      >
        Home
      </Link>
    </span>
    <span
      className="menuItem"
    >
      <Link
        className="clearSesh"
        onClick={[Function]}
        tabIndex="-1"
        to="/"
      >
        CLEAR
      </Link>
    </span>
  </div>
  <div
    id="spaceNav"
  />
</div>
`;

exports[`Header should render tabs with session storage 1`] = `
<div
  id="header"
>
  <div
    id="menuNav"
  >
    <span
      className="menuItem"
    >
      <Link
        tabIndex="-1"
        to="/"
      >
        Home
      </Link>
    </span>
    <span
      className="menuItem"
    >
      <Link
        className="clearSesh"
        onClick={[Function]}
        tabIndex="-1"
        to="/"
      >
        CLEAR
      </Link>
    </span>
  </div>
  <div
    id="spaceNav"
  >
    <span
      className="spaceNavItem "
      key="1"
    >
      <Link
        className=""
        tabIndex="-1"
        to={
          Object {
            "pathname": "/r/",
            "state": Object {
              "pathname": "/s/",
              "state": Object {
                "space": Object {
                  "id": 1,
                  "name": "space1",
                  "status": true,
                },
              },
            },
          }
        }
      >
        space1
      </Link>
    </span>
    <span
      className="spaceNavItem "
      key="2"
    >
      <Link
        className=""
        tabIndex="-1"
        to={
          Object {
            "pathname": "/r/",
            "state": Object {
              "pathname": "/s/",
              "state": Object {
                "space": Object {
                  "id": 2,
                  "name": "space2",
                  "status": true,
                },
              },
            },
          }
        }
      >
        space2
      </Link>
    </span>
    <span
      className="spaceNavItem "
      key="3"
    >
      <Link
        className=""
        tabIndex="-1"
        to={
          Object {
            "pathname": "/r/",
            "state": Object {
              "pathname": "/s/",
              "state": Object {
                "space": Object {
                  "id": 3,
                  "name": "space3",
                  "status": true,
                },
              },
            },
          }
        }
      >
        space3
      </Link>
    </span>
  </div>
</div>
`;
